### ISSUE TITLE
Schema.loadClass overwrite methods from inherited class

### ISSUE THREAD

**Issue by hasezoey** [Automattic/mongoose#8892]
**Do you want to request a *feature* or report a *bug*?**
bug

**What is the current behavior?**
if an class extends another class and has the same names for something like an getter / setter, then the bottom-most class's getter / setter is used

**If the current behavior is a bug, please provide the steps to reproduce.**
(Typescript)
```ts
// NodeJS: 12.16.2
// MongoDB: 4.2-bionic (Docker)
import * as mongoose from "mongoose"; // mongoose@5.9.10

class C1 {
  public get hello() {
    return 1;
  }
}

class C2 extends C1 {
  public get hello() {
    return 2;
  }
}

(async () => {
  await mongoose.connect(`mongodb://localhost:27017/`, { useNewUrlParser: true, dbName: "mongooseLoadClassInheritance", useCreateIndex: true, useUnifiedTopology: true });

  const C1Schema = new mongoose.Schema({});
  C1Schema.loadClass(C1);
  const C1Model = mongoose.model("C1", C1Schema);
  console.log("C1Model", ((new C1Model()) as any).hello); // expected: "1", result: "1"

  const C2Schema = new mongoose.Schema({});
  C2Schema.loadClass(C2);
  const C2Model = mongoose.model("C2", C2Schema);
  console.log("C2Model", ((new C2Model()) as any).hello); // expected: "2", result: "1"

  await mongoose.disconnect();
})();
```

**What is the expected behavior?**
to use the top-most class's getter/setter-

**What are the versions of Node.js, Mongoose and MongoDB you are using? Note that "latest" is not a version.**
mongoose: 5.9.10
typescript: 3.8.3
nodejs: 12.16.2

**Comment by vkarpov15** [comment]
Confirmed that this is already fixed, added a test in #8892

**Comment by hasezoey** [comment]
> Confirmed that this is already fixed, added a test in #8892

for anyone coming across this issue, i think it meant instead of `8892` to be #8897
